src/main/java/frc/robot/subsystems/Drivetrain.java

```diff
@@ -0,0 +1,221 @@
+        kinematics = new DifferentialDriveKinematics(TRACK_WIDTH);
+        ramseteController = new RamseteController();
+        field2d = new Field2d();
+        configureDashboard();
+    }
+
+    @Override
+    public void configureLoop(PIDSettings leftPIDSettings, PIDSettings rightPIDSettings,
+                              FeedForwardSettings feedForwardSettings,
+                              TrapezoidProfileSettings trapezoidProfileSettings) {
+        super.configureLoop(leftPIDSettings, rightPIDSettings, feedForwardSettings, trapezoidProfileSettings);
+        configureEncoders();
+    }
+
+    @Override
+    public void periodic() {
+        super.periodic();
+        odometry.update(gyro.getRotation2d(), getLeftPosition(), getRightPosition());
+        field2d.setRobotPose(getPose2d());
+    }
+
+    public void resetEncoders() {
+        leftEncoder.setPosition(0);
+        rightEncoder.setPosition(0);
+    }
+
+    public void resetOdometry(Pose2d pose2d) {
+        odometry.resetPosition(gyro.getRotation2d(), getLeftPosition(), getRightPosition(), pose2d);
+    }
+
+    public void resetGyro() {
+        gyro.reset();
+    }
+
+    public double getLeftPosition() {
+        return leftEncoder.getPosition();
+    }
+
+    public double getRightPosition() {
+        return rightEncoder.getPosition();
+    }
+
+    public double getPitch() {
+        return gyro.getPitch();
+    }
+
+    public double getYaw() {
+        return gyro.getYaw();
+    }
+
+    public double getLeftSpeed() {
+        return leftEncoder.getVelocity();
+    }
+
+    public double getRightSpeed() {
+        return rightEncoder.getVelocity();
+    }
+
+    public Pose2d getPose2d() {
+        return odometry.getPoseMeters();
+    }
+
+    public PIDSettings getLeftPIDSettings() {
+        return leftPIDSettings;
+    }
+
+    public PIDSettings getRightPIDSettings() {
+        return rightPIDSettings;
+    }
+
+    public PIDSettings getAnglePIDSettings() {
+        return anglePIDSettings;
+    }
+
+    public FeedForwardSettings getFeedForwardSettings() {
+        return feedForwardSettings;
+    }
+
+    public TrapezoidProfileSettings getTrapezoidProfileSettings() {
+        return trapezoidProfileSettings;
+    }
+
+    public DifferentialDriveOdometry getOdometry() {
+        return odometry;
+    }
+
+    public DifferentialDriveKinematics getKinematics() {
+        return kinematics;
+    }
+
+    public RamseteController getRamseteController() {
+        return ramseteController;
+    }
+
+    private void configureEncoders() {
+        leftEncoder.setPositionConversionFactor(DISTANCE_PER_PULSE);
+        leftEncoder.setVelocityConversionFactor(DISTANCE_PER_PULSE / SECONDS_IN_MINUTE);
+        rightEncoder.setPositionConversionFactor(DISTANCE_PER_PULSE);
+        rightEncoder.setVelocityConversionFactor(DISTANCE_PER_PULSE / SECONDS_IN_MINUTE);
+    }
+
+    @Override
+    public void configureDashboard() {
```

Comment:
add x and y

Reply:
had to put in periodic so it will update (`Pose2d` object is changed, so it can't be only once in `configureDashboard()`)